### PR TITLE
Replaced `doc.akka.io` with `pekko.apache.org`

### PR DESCRIPTION
**Pull request by Seetaramayya**
Replaced `doc.akka.io` with `pekko.apache.org`

### CODE REVIEW

**Review by pjfanning — CHANGES_REQUESTED** [review]
We are not ready for this yet. Maybe in a week or 2.

https://pekko.apache.org does not exist yet and we need to agree on the paths we will use.

File: actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/internal/StubbedActorContext.scala [hunk 1]
```diff
@@ -55,7 +55,7 @@ private[pekko] final class FunctionRef[-T](override val path: ActorPath, send: (
   override def provider: ActorRefProvider =
     throw new UnsupportedOperationException(
       "ActorRefs created for synchronous testing cannot be used as targets for asking. Use asynchronous testing instead. " +
-      "See https://doc.akka.io/docs/akka/current/typed/testing.html#asynchronous-testing")
+      "See https://pekko.apache.org/docs/akka/current/typed/testing.html#asynchronous-testing")
```
Comment (pjfanning) [inline actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/internal/StubbedActorContext.scala:58]: there will be no akka in pekko.apache.org paths (ie the `docs/akka` bit) - Pekko team still needs to discuss what paths will end up looking like

Comment (pjfanning) [reply actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/internal/StubbedActorContext.scala:58]: @Seetaramayya I published https://pekko.apache.org/ - if you change the URLs to use `docs/pekko` instead of `docs/akka` then they should work

**Comment by pjfanning** [comment]
@Seetaramayya have you filled in an Apache CLA? We'll need one for a big change like this. See https://github.com/apache/incubator-pekko/Contributing,md

**Comment by Seetaramayya** [comment]
I did not fill Apache CLA agreement yet, I'll do it now.

File: project/Paradox.scala [hunk 2]
```diff
@@ -23,26 +23,27 @@ import sbt._
...
+      "extref.pekko-management.base_url" -> s"$pekkoBaseURL/docs/akka-management/current/%s",
       "extref.platform-guide.base_url" -> "https://developer.lightbend.com/docs/akka-platform-guide/%s",
       "extref.wikipedia.base_url" -> "https://en.wikipedia.org/wiki/%s",
       "extref.github.base_url" -> (GitHub.url(version.value) + "/%s"), // for links to our sources
       "extref.samples.base_url" -> "https://developer.lightbend.com/start/?group=akka&amp;project=%s",
       "extref.ecs.base_url" -> "https://example.lightbend.com/v1/download/%s",
-      "scaladoc.pekko.base_url" -> "https://doc.akka.io/api/akka/2.6/org/apache",
-      "scaladoc.pekko.http.base_url" -> "https://doc.akka.io/api/akka-http/current/org/apache",
+      "pekko.doc.dns" -> s"$pekkoBaseURL",
+      "scaladoc.pekko.base_url" -> s"$pekkoBaseURL/api/akka/2.6/org/apache",
```
Comment (pjfanning) [inline project/Paradox.scala:40]: any path with `/akka` in will need to change to `/pekko`

paths like `akka-http` will need to change to `pekko-http` -- likewise for `akka-management`, etc.

**Review by pjfanning — CHANGES_REQUESTED** [review]

File: persistence/src/main/resources/reference.conf [hunk 3]
```diff
@@ -5,7 +5,8 @@
 # This is the reference config file that contains all the default settings.
 # Make your edits in your application.conf in order to override these settings.
 
-# Directory of persistence journal and snapshot store plugins is available at the 
+# Directory of persistence journal and snapshot store plugins is available at the
+# TODO: Do we need to extend community url of pekko as well along with akka community?
```
Comment (Seetaramayya) [inline persistence/src/main/resources/reference.conf:9]: I don't know what to do, so left `TODO`

File: project/project-info.conf [hunk 4]
```diff
@@ -14,6 +14,7 @@ project-info {
       text: "Github issues"
     }
     release-notes: {
+      # TODO: Correct url needs to be provided
```
Comment (Seetaramayya) [inline project/project-info.conf:17]: Added another `TODO`. Release notes page needs to be added in future. Shall I create a task / issue?

File: cluster-tools/src/main/scala/org/apache/pekko/cluster/client/ClusterClient.scala [hunk 5]
```diff
@@ -56,7 +56,7 @@ import pekko.util.ccompat._
 
 @ccompatUsedUntil213
 @deprecated(
-  "Use Akka gRPC instead, see https://pekko.apache.org/docs/akka/2.6/cluster-client.html#migration-to-akka-grpc",
+  "Use Akka gRPC instead, see https://pekko.apache.org/docs/pekko/2.6/cluster-client.html#migration-to-akka-grpc",
```
Comment (pjfanning) [inline cluster-tools/src/main/scala/org/apache/pekko/cluster/client/ClusterClient.scala:59]: 2.6 is wrong - that is an akka version number - no pekko 2.6 exists

could you check the `migration-to-akka-grpc` bit too? remember that this is pekko, not akka

cluster-client.md may not yet have that header renamed but it should be - could you check?

File: docs/src/main/paradox/cluster-client.md [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 @@@ warning
 
-Cluster Client is deprecated in favor of using [Pekko gRPC](https://doc.akka.io/docs/akka-grpc/current/index.html).
+Cluster Client is deprecated in favor of using [Pekko gRPC]($pekko.doc.dns$/docs/akka-grpc/current/index.html).
```
Comment (pjfanning) [inline docs/src/main/paradox/cluster-client.md:5]: akka-grpc is renamed to pekko-grpc

File: docs/src/main/paradox/cluster-client.md [hunk 7]
```diff
@@ -38,8 +38,8 @@ i.e. not necessarily the initial contact points.
 Using the @apidoc[ClusterClient] for communicating with a cluster from the outside requires that the system with the client
 can both connect and be connected with Pekko Remoting from all the nodes in the cluster with a receptionist.
 This creates a tight coupling in that the client and cluster systems may need to have the same version of both Pekko, libraries, message classes, serializers and potentially even the JVM. In many cases it is a better solution
-to use a more explicit and decoupling protocol such as [HTTP](https://doc.akka.io/docs/akka-http/current/index.html) or
-[gRPC](https://doc.akka.io/docs/akka-grpc/current/).
+to use a more explicit and decoupling protocol such as [HTTP]($pekko.doc.dns$/docs/akka-http/current/index.html) or
+[gRPC]($pekko.doc.dns$/docs/akka-grpc/current/).
```
Comment (pjfanning) [inline docs/src/main/paradox/cluster-client.md:42]: pekko-http

File: persistence/src/main/resources/reference.conf (hunk 3 above)
Comment (pjfanning) [reply persistence/src/main/resources/reference.conf:9]: please remove the TODO - the fact that the akka url is there will attract attention

it's ok to leave the https://akka.io/community/ as is until we work out what to replace it with

File: actor/src/main/scala/org/apache/pekko/io/Udp.scala [hunk 8]
```diff
@@ -36,7 +36,7 @@ import pekko.util.ccompat._
  * from whom data can be received. For “connected” UDP mode see [[UdpConnected]].
  *
  * For a full description of the design and philosophy behind this IO
- * implementation please refer to <a href="https://akka.io/docs/">the Akka online documentation</a>.
+ * implementation please refer to <a href="https://pekko.apache.org/docs/pekko/current/">the Akka online documentation</a>.
```
Comment (pjfanning) [inline actor/src/main/scala/org/apache/pekko/io/Udp.scala:39]: Could you replace 'the Akka online documentation'?

File: docs/src/main/paradox/common/other-modules.md [hunk 9]
```diff
@@ -1,62 +1,62 @@
...
+## [Pekko HTTP]($pekko.doc.dns$/docs/pekko-http/current/)
 
 A full server- and client-side HTTP stack on top of pekko-actor and pekko-stream.
 
-## [Pekko gRPC](https://doc.akka.io/docs/akka-grpc/current/)
+## [Pekko gRPC]($pekko.doc.dns$/docs/pekko-grpc/current/)
 
 Pekko gRPC provides support for building streaming gRPC servers and clients on top of Pekko Streams.
 
-## [Pekko Connectors](https://doc.akka.io/docs/alpakka/current/)
```
Comment (pjfanning) [inline docs/src/main/paradox/common/other-modules.md:13]: alpakka is wrong - should be pekko-connectors

File: docs/src/main/paradox/common/other-modules.md [hunk 10]
```diff
@@ -1,62 +1,62 @@
...
+## [Pekko Connectors]($pekko.doc.dns$/docs/alpakka/current/)
 
 Pekko Connectors is a Reactive Enterprise Integration library for Java and Scala, based on Reactive Streams and Pekko.
 
-## [Pekko Kafka Connector](https://doc.akka.io/docs/alpakka-kafka/current/)
+## [Pekko Kafka Connector]($pekko.doc.dns$/docs/alpakka-kafka/current/)
```
Comment (pjfanning) [inline docs/src/main/paradox/common/other-modules.md:17]: alpakka-kafka is wrong - should be pekko-connectors-kafka

File: cluster-tools/src/main/scala/org/apache/pekko/cluster/client/ClusterClient.scala (hunk 5 above)
Comment (Seetaramayya) [reply cluster-tools/src/main/scala/org/apache/pekko/cluster/client/ClusterClient.scala:59]: Whole migration section can be deleted and reference `pekko-grpc` link

File: project/Paradox.scala [hunk 11]
```diff
@@ -23,26 +23,27 @@ import sbt._
...
-      "canonical.base_url" -> "https://doc.akka.io/docs/akka/current",
+      "canonical.base_url" -> s"$pekkoBaseURL/docs/pekko/current",
       "github.base_url" -> GitHub
         .url(version.value), // for links like this: @github[#1](#1) or @github[83986f9](83986f9)
-      "extref.pekko.http.base_url" -> "https://doc.akka.io/docs/akka-http/current/%s",
-      "extref.pekko-management.base_url" -> "https://doc.akka.io/docs/akka-management/current/%s",
-      "extref.platform-guide.base_url" -> "https://developer.lightbend.com/docs/akka-platform-guide/%s",
+      "extref.pekko.http.base_url" -> s"$pekkoBaseURL/docs/pekko-http/current/%s",
+      "extref.pekko-management.base_url" -> s"$pekkoBaseURL/docs/pekko-management/current/%s",
+      "extref.platform-guide.base_url" -> "https://developer.lightbend.com/docs/pekko-platform-guide/%s",
```
Comment (pjfanning) [inline project/Paradox.scala:34]: change the host to $pekkoBaseURL/ - we'll host the pekko-platform-guide there

File: project/Publish.scala [hunk 12]
```diff
@@ -49,6 +49,7 @@ object Publish extends AutoPlugin {
     val key = new java.io.File(
       Option(System.getProperty("pekko.gustav.key"))
         .getOrElse(System.getProperty("user.home") + "/.ssh/id_rsa_gustav.pem"))
+    // TODO: gustav.pekko.io?
```
Comment (pjfanning) [inline project/Publish.scala:52]: remove the TODO, we'll sort out the rsync stuff later

File: project/project-info.conf (hunk 4 above)
Comment (pjfanning) [reply project/project-info.conf:17]: remove the TODO, this problem URL is easily found when we come back to remove the final akka refs

File: remote/src/test/scala/org/apache/pekko/remote/Ticket1978CommunicationSpec.scala [hunk 13]
```diff
@@ -193,7 +193,7 @@ abstract class Ticket1978CommunicationSpec(val cipherConfig: CipherConfig)
 
         val compressed = baos.toByteArray
         // random data should not be compressible
-        // Another reproducer of https://doc.akka.io/docs/akka/current/security/2018-08-29-aes-rng.html
+        // Another reproducer of https://pekko.apache.org/docs/pekko/current/security/2018-08-29-aes-rng.html
```
Comment (pjfanning) [inline remote/src/test/scala/org/apache/pekko/remote/Ticket1978CommunicationSpec.scala:196]: revert this - we are not keeping the akka security docs in the pekko doc clone - so best to link this to the akka docs

File: remote/src/test/scala/org/apache/pekko/remote/artery/tcp/SecureRandomFactorySpec.scala [hunk 14]
```diff
@@ -51,7 +51,7 @@ abstract class SecureRandomFactorySpec(alg: String) extends PekkoSpec {
       "generate random" in {
         val bytes = Array.ofDim[Byte](16)
         // Reproducer of the specific issue described at
-        // https://doc.akka.io/docs/akka/current/security/2018-08-29-aes-rng.html
+        // https://pekko.apache.org/docs/pekko/current/security/2018-08-29-aes-rng.html
```
Comment (pjfanning) [inline remote/src/test/scala/org/apache/pekko/remote/artery/tcp/SecureRandomFactorySpec.scala:54]: revert this - we are not keeping the akka security docs in the pekko doc clone - so best to link this to the akka docs

File: remote/src/test/scala/org/apache/pekko/remote/artery/tcp/SecureRandomFactorySpec.scala [hunk 15]
```diff
@@ -75,7 +75,7 @@ abstract class SecureRandomFactorySpec(alg: String) extends PekkoSpec {
 
         val compressed = baos.toByteArray
         // random data should not be compressible
-        // Another reproducer of https://doc.akka.io/docs/akka/current/security/2018-08-29-aes-rng.html
+        // Another reproducer of https://pekko.apache.org/docs/pekko/current/security/2018-08-29-aes-rng.html
```
Comment (pjfanning) [inline remote/src/test/scala/org/apache/pekko/remote/artery/tcp/SecureRandomFactorySpec.scala:78]: revert this - we are not keeping the akka security docs in the pekko doc clone - so best to link this to the akka docs

File: stream/src/main/scala/org/apache/pekko/stream/ActorMaterializer.scala [hunk 16]
```diff
@@ -223,7 +223,7 @@ private[pekko] object ActorMaterializerHelper {
 abstract class ActorMaterializer extends Materializer with MaterializerLoggingProvider {
 
   @deprecated(
-    "Use attributes to access settings from stages, see https://doc.akka.io/docs/akka/2.6/project/migration-guide-2.5.x-2.6.x.html",
+    "Use attributes to access settings from stages, see https://pekko.apache.org/docs/pekko/2.6/project/migration-guide-2.5.x-2.6.x.html",
```
Comment (pjfanning) [inline stream/src/main/scala/org/apache/pekko/stream/ActorMaterializer.scala:226]: we have removed the akka migration guides - so best to revert this change so it links to the akka docs

Comment (pjfanning) [reply stream/src/main/scala/org/apache/pekko/stream/ActorMaterializer.scala:226]: all other changes like this should be reverted too

**Comment by Seetaramayya** [comment]
What to do?
Still I see some references to `lightbend` guides, for example 
```
https://developer.lightbend.com/guides/akka-quickstart-scala/
```

in `exampels.md`, `introduction.md`, `tutorial.md` and  `tutorial_1.md`

**Comment by pjfanning** [comment]
> What to do? Still I see some references to `lightbend` guides, for example
> 
> ```
> https://developer.lightbend.com/guides/akka-quickstart-scala/
> ```
> 
> in `exampels.md`, `introduction.md`, `tutorial.md` and `tutorial_1.md`

we can't fix everything today - I've raised https://github.com/apache/incubator-pekko/issues/123 to come back to this bit

File: project/Paradox.scala [hunk 17]
```diff
@@ -23,26 +23,27 @@ import sbt._
...
         .url(version.value), // for links like this: @github[#1](#1) or @github[83986f9](83986f9)
-      "extref.pekko.http.base_url" -> "https://doc.akka.io/docs/akka-http/current/%s",
-      "extref.pekko-management.base_url" -> "https://doc.akka.io/docs/akka-management/current/%s",
-      "extref.platform-guide.base_url" -> "https://developer.lightbend.com/docs/akka-platform-guide/%s",
+      "extref.pekko.http.base_url" -> s"$pekkoBaseURL/docs/pekko-http/current/%s",
+      "extref.pekko-management.base_url" -> s"$pekkoBaseURL/docs/pekko-management/current/%s",
+      "extref.platform-guide.base_url" -> s"$pekkoBaseURL/",
       "extref.wikipedia.base_url" -> "https://en.wikipedia.org/wiki/%s",
       "extref.github.base_url" -> (GitHub.url(version.value) + "/%s"), // for links to our sources
       "extref.samples.base_url" -> "https://developer.lightbend.com/start/?group=akka&amp;project=%s",
```
Comment (Seetaramayya) [inline project/Paradox.scala:37]: I don't think `extref.samples.base_url` exist. This results `404`s on lightbend side because `.*pekko.*` project does not exists on their side. For example, 

```
https://developer.lightbend.com/start/?group=akka&project=pekko-sample-cluster-docker-compose-scala
```

Comment (pjfanning) [reply project/Paradox.scala:37]: just add issues for anything you are skipping

Comment (Seetaramayya) [reply project/Paradox.scala:37]: Added an issue https://github.com/apache/incubator-pekko/issues/124

**Comment by Seetaramayya** [comment]
> I did not fill Apache CLA agreement yet, today I'll do it.

Done, I got confirmation that `ICLA has been filed`

File: cluster-tools/src/main/scala/org/apache/pekko/cluster/client/ClusterClient.scala [hunk 18]
```diff
@@ -127,7 +127,7 @@ object ClusterClientSettings {
  *   external service registry
  */
 @deprecated(
-  "Use Akka gRPC instead, see https://doc.akka.io/docs/akka/2.6/cluster-client.html#migration-to-akka-grpc",
+  "Use Akka gRPC instead, see https://pekko.apache.org/docs/pekko/2.6/cluster-client.html#migration-to-akka-grpc",
```
Comment (pjfanning) [inline cluster-tools/src/main/scala/org/apache/pekko/cluster/client/ClusterClient.scala:130]: can you fix up the rest of this class - should not have `pekko/2.6` - use `pekko/current`

File: docs/src/main/paradox/.htaccess [hunk 19]
```diff
@@ -18,9 +18,9 @@ RedirectMatch 301 ^(.*[^d])/guide/(.*) $1/typed/guide/$2
 RedirectMatch 301 ^(.*)/additional/rolling-deploys\.html$ $1/additional/rolling-updates.html
 
 RedirectMatch 301 ^(.*)/additional/index\.html$ $1/project/index.html
-RedirectMatch 301 ^(.*)/howto\.html$ https://doc.akka.io/docs/akka/2.5/howto.html
-RedirectMatch 301 ^(.*)/common/duration\.html$ https://doc.akka.io/docs/akka/2.5/common/duration.html
-RedirectMatch 301 ^(.*)/java8-compat\.html$ https://doc.akka.io/docs/akka/2.5/java8-compat.html
+RedirectMatch 301 ^(.*)/howto\.html$ https://pekko.apache.org/docs/pekko/2.5/howto.html
```
Comment (pjfanning) [inline docs/src/main/paradox/.htaccess:21]: remove the redirects that mention pekko/2.5 - we don't need these redirects

File: docs/src/main/paradox/cluster-client.md [hunk 20]
```diff
@@ -295,7 +295,7 @@ based API.
 
 ##### Initial contact points
 
-Instead of configured initial contact points the [Pekko gRPC Service Discovery](https://doc.akka.io/docs/akka-grpc/current/client/configuration.html#using-akka-discovery-for-endpoint-discovery) can be used.
+Instead of configured initial contact points the [Pekko gRPC Service Discovery]($pekko.doc.dns$/docs/pekko-grpc/current/client/configuration.html#using-akka-discovery-for-endpoint-discovery) can be used.
```
Comment (pjfanning) [inline docs/src/main/paradox/cluster-client.md:299]: pekko-discovery

Comment (Seetaramayya) [reply docs/src/main/paradox/cluster-client.md:299]: As far as I understand, we need to refer https://github.com/apache/incubator-pekko-grpc/blob/main/docs/src/main/paradox/client/configuration.md#using-akka-discovery-for-endpoint-discovery from this location, this should be access by `pekk-grpc` right? 

or Am I miss understanding something?

Comment (pjfanning) [reply docs/src/main/paradox/cluster-client.md:299]: `#using-akka-discovery-for-endpoint-discovery` should be `#using-pekko-discovery-for-endpoint-discovery`

Comment (Seetaramayya) [reply docs/src/main/paradox/cluster-client.md:299]: Okay, I'll change it. `#` link will not work until we change in `incubator-pekko-grpc` project

File: project/Doc.scala [hunk 21]
```diff
@@ -74,7 +74,7 @@ object Scaladoc extends AutoPlugin {
       "-doc-version",
       ver,
       "-doc-canonical-base-url",
-      "https://doc.akka.io/api/akka/current/")
```
Comment (pjfanning) [inline project/Doc.scala:77]: A couple of lines above this - could you replace this?

```
      "-doc-title",
      "Akka",
```
with
```
      "-doc-title",
      "Apache Pekko",
```

Comment (Seetaramayya) [reply project/Doc.scala:77]: Done

File: project/Publish.scala [hunk 22]
```diff
@@ -35,10 +35,10 @@ object Publish extends AutoPlugin {
     startYear := Some(2009),
     developers := List(
       Developer(
-        "akka-contributors",
-        "Akka Contributors",
+        "pekko-contributors",
+        "Pekko Contributors",
         "akka.official@gmail.com",
```
Comment (Seetaramayya) [inline project/Publish.scala:40]: What is the equivalent email address for `akka.official@gmail.com` ?

Comment (pjfanning) [reply project/Publish.scala:40]: use `dev@pekko.apache.org`

Comment (Seetaramayya) [reply project/Publish.scala]: Ignore my previous comment, just now saw on main it is modified

**Comment by pjfanning** [comment]
@Seetaramayya have you started the process of getting the CLA approved?

**Comment by Seetaramayya** [comment]
@pjfanning Is it sufficient?
image

**Comment by pjfanning** [comment]
@Seetaramayya could you tell me the email address that you used to get the iCLA - because I can't find your iCLA in the ASF app and it is based on email addresses?

**Review by pjfanning — APPROVED** [review]
I found the iCLA - approved
